src/main/java/appeng/client/render/items/ToolWirelessTerminalRender.java

```diff
@@ -0,0 +1,135 @@
+package appeng.client.render.items;
+
+import appeng.api.util.AEColor;
+import appeng.client.texture.ExtraItemTextures;
+import appeng.items.tools.powered.ToolWirelessTerminal;
+import buildcraft.core.lib.engines.RenderEngine;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.renderer.ItemRenderer;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.entity.RenderItem;
+import net.minecraft.client.renderer.texture.ITextureObject;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.IItemRenderer;
+import org.lwjgl.opengl.GL11;
+
+import javax.swing.*;
+
+/**
+ * Created by Tom on 9/30/2015.
+ */
+public class ToolWirelessTerminalRender implements IItemRenderer {
+
+    @Override
+    public boolean handleRenderType( ItemStack item, ItemRenderType type ) {
+        return true;
+    }
+
+    @Override
+    public boolean shouldUseRenderHelper( ItemRenderType type, ItemStack item, ItemRendererHelper helper ) {
+        return false;
+    }
+
+    @Override
+    public void renderItem( ItemRenderType type, ItemStack item, Object... data ) {
+        // Get icon index for the texture
+        IIcon icon = ExtraItemTextures.WirelessTerminal_Inactive.getIcon();
+        if( ToolWirelessTerminal.isLinked( item ) )
+            icon = item.getIconIndex();
+
+        AEColor color = ToolWirelessTerminal.getColor( item );
+        float r = 0;
+        float g = 0;
+        float b = 0;
+        if( color!=null ) {
+            int medColor = color.mediumVariant;
+            r = ( medColor >> 16 ) & 0xFF;
+            g = ( medColor >> 8 ) & 0xFF;
+            b = ( medColor >> 0 ) & 0xFF;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this useless shift [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2183)